bundles/net.tourbook/src/net/tourbook/tag/Dialog_TourTag.java

```diff
@@ -165,15 +246,88 @@ protected void okPressed() {
       super.okPressed();
    }
 
+   private void onDeleteImage() {
+
+      _imageFilePath = null;
+
+      setTagImage(null);
+
+      enableControls();
+   }
+
+   private void onDispose() {
+
+      if (_canvasTagImage == null ||
+            _canvasTagImage.getImage() == null ||
+            _canvasTagImage.isDisposed()) {
+         return;
+      }
+
+      _canvasTagImage.dispose();
+   }
+
+   private void onSelectImage() {
+
+      final FileDialog fileDialog = new FileDialog(getShell(), SWT.OPEN);
+
+      fileDialog.setText(Messages.Dialog_TourTag_ImportImage_Title);
+      fileDialog.setFilterPath(_prefStore.getString(IMAGE_LAST_SELECTED_PATH));
+      fileDialog.setFilterNames(new String[] { Messages.Dialog_TourTag_FileDialog_ImageFiles });
+      fileDialog.setFilterExtensions(new String[] { "*.bmp;*.gif;*.png;*.jpg" });//$NON-NLS-1$
+
+      // open file dialog
+      final String imageFilePath = fileDialog.open();
+
+      // check if user canceled the dialog
+      if (imageFilePath == null) {
+         return;
+      }
+
+      setTagImage(imageFilePath);
+
+      final String filePathFolder = Paths.get(imageFilePath).getParent().toString();
+
+      // keep last selected path
+      _prefStore.putValue(IMAGE_LAST_SELECTED_PATH, filePathFolder);
+
+      enableControls();
+   }
+
    private void restoreState() {
 
       _txtName.setText(_tourTag_Clone.getTagName());
       _txtNotes.setText(_tourTag_Clone.getNotes());
+      _imageFilePath = _tourTag_Clone.getImageFilePath();
+      setTagImage(_imageFilePath);
    }
 
    private void saveState() {
 
       _tourTag_Clone.setNotes(_txtNotes.getText());
       _tourTag_Clone.setTagName(_txtName.getText());
+      _tourTag_Clone.setImageFilePath(_imageFilePath);
+   }
+
+   private void setTagImage(final String imageFilePath) {
+
+      Image image =
+            TourbookPlugin.getImageDescriptor(net.tourbook.Images.Camera).createImage();
+      _lblInvalidImageError.setText(UI.EMPTY_STRING);
+
+      if (StringUtils.hasContent(imageFilePath)) {
+
+         if (!Files.exists(Paths.get(imageFilePath))) {
+
+            _lblInvalidImageError.setText(NLS.bind(
+                  Messages.Dialog_TourTag_Label_ImageNotFound,
+                  imageFilePath));
```

Comment:
A messages dialog already have an error tracking system, an error can be set with `org.eclipse.jface.dialogs.TitleAreaDialog.setErrorMessage(String)` and is used several times in MT

![image](https://user-images.githubusercontent.com/1283445/218697722-83e1c569-757f-4852-95b9-d3b87b028aee.png)

Reply:
Good idea, I forgot about this one.
Fixed:
![image](https://user-images.githubusercontent.com/25235665/218874143-a10c1d27-b7cc-453f-8e76-e7b1d7d25db1.png)

Reply:
#### Enhancement
I think it would look better, when these 2 buttons are aligned to the right side. So the image should grab all free horizontal space but is aligned to the left side

Reply:
Fixed

![image](https://user-images.githubusercontent.com/25235665/219204907-9be0b58b-8983-4fb0-a3ad-4c1ca7716bce.png)